src/dotnet/APIView/APIViewWeb/LeanControllers/UserProfileController.cs

```diff
@@ -11,33 +13,38 @@ public class UserProfileController : BaseApiController
     {
         private readonly ILogger<AuthController> _logger;
         private readonly IUserProfileManager _userProfileManager;
-        private readonly UserPreferenceCache _userPreferenceCache;
+        private readonly UserProfileCache _userProfileCache;
 
-        public UserProfileController(ILogger<AuthController> logger, IUserProfileManager userProfileManager,
-            UserPreferenceCache userPreferenceCache)
+        public UserProfileController(ILogger<AuthController> logger, IUserProfileManager userProfileManager, UserProfileCache userProfileCache)
         {
             _logger = logger;
             _userProfileManager = userProfileManager;
-            _userPreferenceCache = userPreferenceCache;
+            _userProfileCache = userProfileCache;
         }
 
         [HttpGet]
-        public async Task<ActionResult<UserProfileModel>> GetUserPreference()
+        public async Task<ActionResult<UserProfileModel>> GetUserPreference([FromQuery]string userName = null)
         {
-            var userProfile = await _userProfileManager.TryGetUserProfileAsync(User);
-            var preference = await _userPreferenceCache.GetUserPreferences(User);
-            if (preference != null)
+            userName = userName ?? User.GetGitHubLogin();
+            var userProfile = await _userProfileCache.GetUserProfileAsync(userName, createIfNotExist: false);
+            if (userProfile == default)
             {
-                userProfile.Preferences = preference;
+                return new LeanJsonResult(null, StatusCodes.Status404NotFound);
             }
-            return userProfile;
+            return new LeanJsonResult(userProfile, StatusCodes.Status200OK);
         }
 
-        [Route("preference")]
-        [HttpPut]
-        public ActionResult UpdateUserPreference([FromBody] UserPreferenceModel userPreference)
+        [HttpPut("preference", Name = "UpdateUserPreference")]
+        public async Task<ActionResult> UpdateUserPreference([FromBody] UserPreferenceModel userPreference)
         {
-            _userPreferenceCache.UpdateUserPreference(userPreference, User);
+            await _userProfileCache.UpdateUserProfileAsync(userName: User.GetGitHubLogin(), userPreferenceModel: userPreference);
+            return Ok();
```

Comment:
what happens if `_userProfileCache.UpdateUserProfileAsync` fails? how is the exception and error message handled?

Reply:
IT will return 500, `InternalServerError` to the caller